### PR TITLE
scripts: add generate_cloud_provider_config.sh

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
**Problem:**
Now, creating Harvester CSI driver w/o cloud provider will miss the cloud-config even following the steps on the document.

**Solution:**
we need to generate the cloud-config when using on RKE2/K3S w/o cloud provider

**Related Issue:**
This is required with the document update: https://github.com/harvester/docs/pull/309

**Test plan:**
Create Harvester CSI driver w/o cloud provider successfully on RKE2/K3S

### CODE REVIEW

**Review by guangbochen — COMMENTED** [review]
can we re-use or merge this with https://github.com/harvester/harvester-csi-driver/blob/master/deploy/generate_addon_csi.sh?

**Comment by Vicente-Cheng** [comment]
> can we re-use or merge this with https://github.com/harvester/harvester-csi-driver/blob/master/deploy/generate_addon_csi.sh?

Do we still need to generate_addon_csi.sh to generate the addon.yaml for rke1?
If not, we could rewrite this script. Or we could add another argument for generating yaml for different purposes.

File: deploy/generate_addon_csi.sh [hunk 1]
```diff
@@ -143,11 +150,37 @@ $kubeconfig
...
 extract_ca_crt_from_secret
 get_user_token_from_secret
 set_kube_config_values
-assemble_addon_config
+
+if [[ $SCRIPT_TARGET == "RKE1" ]]; then
+  assemble_addon_config
+elif [[ $SCRIPT_TARGET == "RKE2" || $SCRIPT_TARGET == "K3S" ]]; then
+  generate_cloud_config
+else
```
Comment (guangbochen) [inline deploy/generate_addon_csi.sh:184]: can we validate it using case insensitive or lower-case only

File: deploy/generate_addon_csi.sh [hunk 2]
```diff
@@ -2,13 +2,20 @@
 set -e
 set -o pipefail
 
+echo "NOTE: your are using RKE2/K3S, please follow this usage: $0 <service_account_name> <namespace> RKE2/K3S"
```
Comment (guangbochen) [inline deploy/generate_addon_csi.sh:5]:
```suggestion
echo "NOTE: if your are using RKE2/K3S, please follow this usage: $0 <service_account_name> <namespace> rke2/k3s"
```

File: deploy/generate_addon_csi.sh [hunk 3]
```diff
@@ -143,11 +150,37 @@ $kubeconfig
...
 get_user_token_from_secret
 set_kube_config_values
-assemble_addon_config
+
+if [[ $SCRIPT_TARGET == "RKE1" ]]; then
+  assemble_addon_config
+elif [[ $SCRIPT_TARGET == "RKE2" || $SCRIPT_TARGET == "K3S" ]]; then
+  generate_cloud_config
+else
+  echo "Not Supported TARGET $SCRITP_TARGET, please use RKE1, RKE2 or K3S"
```
Comment (guangbochen) [inline deploy/generate_addon_csi.sh:185]:
```suggestion
  echo "Target $SCRITP_TARGET is not supported, please use RKE1, RKE2 or K3S"
```

File: deploy/generate_addon_csi.sh (hunk 2 above)
Comment (bk201) [inline deploy/generate_addon_csi.sh:5]:
```suggestion
echo "NOTE: if you are using RKE2/K3S, please follow this usage: $0 <service_account_name> <namespace> RKE2/K3S"
```

File: deploy/generate_addon_csi.sh (hunk 1 above)
Comment (bk201) [reply deploy/generate_addon_csi.sh:184]: Perhaps verifying on line 18 to fail a bad input early without creating those secrets.

File: deploy/generate_addon_csi.sh (hunk 2 above)
Comment (Vicente-Cheng) [reply deploy/generate_addon_csi.sh]: done.

File: deploy/generate_addon_csi.sh (hunk 1 above)
Comment (Vicente-Cheng) [reply deploy/generate_addon_csi.sh]: Updated with both suggestions. Please help to recheck it.

**Review by guangbochen — APPROVED** [review]
LGTM, thanks

**Pull request merged**